js/src/components/paid-ads/ads-campaign/ads-campaign.js

```diff
@@ -0,0 +1,182 @@
+ * so it expects a `CampaignAssetsForm` to existing in its parents.
+ *
+ * @fires gla_documentation_link_click with `{ context: 'create-ads' | 'edit-ads' | 'setup-ads', link_id: 'see-what-ads-look-like', href: 'https://support.google.com/google-ads/answer/6275294' }`
+ * @fires gla_onboarding_complete_with_paid_ads_button_click
+ * @param {Object} props React props.
+ * @param {Campaign} [props.campaign] Campaign data to be edited. If not provided, this component will show campaign creation UI.
+ * @param {string} props.headerTitle The title of the step.
+ * @param {string} [props.headerDescription] The description of the step.
+ * @param {() => void} props.onContinue Callback called once continue button is clicked.
+ * @param {() => void} [props.onSkip] Callback called once skip button is clicked.
+ * @param {boolean} [props.hasError=false] Whether there's an error to reset the completing state.
+ * @param {boolean} [props.isOnboardingFlow=false] Whether this component is used in onboarding flow.
+ * @param {'create-ads'|'edit-ads'|'setup-ads'} props.trackingContext A context indicating which page this component is used on. This will be the value of `context` in the track event properties.
+ */
+export default function AdsCampaign( {
+	campaign,
+	headerTitle,
+	headerDescription,
+	onContinue,
+	onSkip = noop,
+	hasError = false,
+	isOnboardingFlow = false,
+	trackingContext,
+} ) {
+	const formContext = useAdaptiveFormContext();
+	const { data: countryCodes } = useTargetAudienceFinalCountryCodes();
+	const { isValidForm, setValue } = formContext;
+	const [ completing, setCompleting ] = useState( null );
+	const [ paidAds, setPaidAds ] = useState( {} );
+
+	useEffect( () => {
+		if ( hasError ) {
+			setCompleting( null );
+		}
+	}, [ hasError ] );
+
+	const handleOnStatesReceived = ( paidAdsValues ) => {
+		setPaidAds( paidAdsValues );
+
+		const { amount } = paidAdsValues;
+		setValue( 'amount', amount );
+	};
+
+	const handleSkipCreateAds = () => {
+		setCompleting( ACTION_SKIP );
+
+		onSkip( paidAds );
+	};
+
+	const handleCompleteClick = ( event ) => {
+		setCompleting( event.target.dataset.action );
+
+		onContinue( paidAds );
+	};
+
+	// The status check of Google Ads account connection is included in `paidAds.isReady`,
+	// because when there is no connected account, it will disable the budget section and set the `amount` to `undefined`.
+	const disabledComplete =
+		completing === ACTION_SKIP || ! paidAds.isReady || ! isValidForm;
+
+	let continueButtonProps = {
+		text: __( 'Continue', 'google-listings-and-ads' ),
+	};
+
+	if ( isOnboardingFlow ) {
+		continueButtonProps = {
+			'data-action': ACTION_COMPLETE,
+			text: __( 'Complete setup', 'google-listings-and-ads' ),
+			eventName: 'gla_onboarding_complete_with_paid_ads_button_click',
+			eventProps: {
+				budget: paidAds.amount,
+				audiences: countryCodes?.join( ',' ),
+			},
+		};
+	}
+
+	const description =
+		headerDescription ||
```

Comment:
`headerDescription` seems to be a required property, should it default to `null` or `''` to allow it to be empty? What if someone intentionally passes a falsey value to that prop? Won't this fallback kick in? Similarly, we don't seem to handle any default fallback for the `headerTitle` prop here. Is that intentional?

Reply:
@joemcgill I removed the `headerDescription` prop and instead conditionally set the description based on the `isOnboardingFlow` prop. Because it's only during the onboarding flow that the description is different.